src/main/java/bio/terra/cli/command/notebooks/Create.java

```diff
@@ -92,11 +100,33 @@ protected void execute() {
     envVars.put("SUBNET", "projects/" + projectId + "/regions/" + zone + "/subnetworks/subnetwork");
     envVars.put("TERRA_WORKSPACE_ID", workspaceContext.getWorkspaceId().toString());
 
+    OUT.println(
+        String.format(
+            "Creating notebook instance `projects/%s/locations/%s/instanceId/%s`",
+            projectId, location, instanceName));
     new DockerAppsRunner(globalContext, workspaceContext).runToolCommand(command, envVars);
 
     OUT.println(
         "Notebook instance starting. This will take ~5-10 minutes.\n"
             + "See your notebooks in this workspace at https://console.cloud.google.com/ai-platform/notebooks/list/instances?project="
             + projectId);
   }
+
+  /**
+   * Returns the specified instanceName or an auto generated instance name with the username and
+   * date time.
+   */
+  private String getInstanceName(TerraUser user) {
+    if (!AUTO_GENERATE_NAME.equals(rawInstanceName)) {
+      return rawInstanceName;
+    }
+    String localDateTimeSuffix =
+        DateTimeFormatter.ofPattern(AUTO_NAME_DATE_FORMAT)
+            .format(Instant.now().atZone(ZoneId.systemDefault()));
+    return extractUsername(user.terraUserEmail) + localDateTimeSuffix;
+  }
+
+  private static String extractUsername(String validEmail) {
+    return validEmail.substring(0, validEmail.indexOf('@'));
```

Comment:
Will this work for an email such as `mariko.medlock@gmail.com`? Asking because the option description seems to mean that `.` is not allowed.

Reply:
Good point. Took a stab at handling more email addresses, PTAL.